webp_backport/src/main/java/de/marcreichelt/webp_backport/FileSignatureChecker.java

```diff
@@ -0,0 +1,59 @@
+package de.marcreichelt.webp_backport;
+
+class FileSignatureChecker {
+
+    /**
+     * Check if a given byte array has the necessary WebP signature in the beginning of its content
+     *
+     * @param imgByteArr the byte array to check
+     * @return true if the byte array has the WebP signature else false
+     */
+    static boolean checkForWebP(byte[] imgByteArr) {
+        if (byteArrayStartsWith(imgByteArr, new byte[]{82, 73, 70, 70})) { // bytes 1 to 3 must be 'RIFF'
```

Comment:
Comment must be "bytes 0 to 3 must be 'RIFF'" (off-by-one error).
Or better: // starts with 'RIFF'